File: src/packets/PacketC2SZoneMaxVolume.js
const PacketC2S = require("./PacketC2S");

/**
 * Client -> Server
 * ID = 0x64
 * Zone Max Volume
 * Sets a zone's max volume
 * Data:
 *      (Unsigned Char) Controller ID
 *      (Unsigned Char) Zone ID
 *      (Unsigned Char) Max Volume
 */
class PacketC2SZoneMaxVolume extends PacketC2S {
    parseData() {
        this._ctrllrID = this._buffer.readUInt8();
        this._zoneID = this._buffer.readUInt8();
        this._maxVolume = this._buffer.readUInt8();
    }

    getControllerID() {
        return this._ctrllrID;
    }

    getZoneID() {
        return this._zoneID;
    }

    getMaxVolume() {
        return this._maxVolume;
    }

    getID() {
        return PacketC2SZoneMaxVolume.ID;
    }
}

PacketC2SZoneMaxVolume.ID = 0x64;

module.exports = PacketC2SZoneMaxVolume;

File: src/RNetPi.Core/Packets/PacketC2SZoneMaxVolume.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Client -> Server
/// ID = 0x64
/// Zone Max Volume
/// Sets a zone's max volume
/// Data:
///     (Unsigned Char) Controller ID
///     (Unsigned Char) Zone ID
///     (Unsigned Char) Max Volume
/// </summary>
public class PacketC2SZoneMaxVolume : PacketC2S
{
    public const byte ID = 0x64;

    public byte ControllerID { get; private set; }
    public byte ZoneID { get; private set; }
    public byte MaxVolume { get; private set; }

    public PacketC2SZoneMaxVolume(byte[] data) : base(data)
    {
    }

    public override byte GetID() => ID;

    protected override void ParseData()
    {
        ControllerID = Reader.ReadByte();
        ZoneID = Reader.ReadByte();
        MaxVolume = Reader.ReadByte();
    }
}